### PR TITLE
feat: add data fetcher service

### PR DESCRIPTION
**Pull request by vasyl-ivanchuk**
# What ❔

Add data fetcher service that can be vertically scaled.

## Why ❔

Having data fetching vertically scaled we will be able to keep up with higher TPS.

## Checklist

- [X] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [X] Tests for the changes have been added / updated.

### CODE REVIEW

File: packages/data-fetcher/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# zkSync Era Block Explorer Data Fetcher
```
Comment (Romsters) [inline packages/data-fetcher/README.md:1]: TODO: add readme

File: packages/data-fetcher/src/address/address.service.ts [hunk 2]
```diff
@@ -61,8 +42,15 @@ export class AddressService {
...
 
-    this.logger.debug({ message: "Saving contract addresses.", transactionReceipt: transactionReceipt.blockNumber });
-    await Promise.all(contractAddresses.map((contractAddress) => this.saveContractAddress(contractAddress)));
+    this.logger.debug({
+      message: "Requesting contracts' bytecode",
+      transactionReceipt: transactionReceipt.blockNumber,
+    });
+    await Promise.all(
+      contractAddresses.map(async (contractAddress) => {
+        contractAddress.bytecode = await this.blockchainService.getCode(contractAddress.address);
```
Comment (Romsters) [inline packages/data-fetcher/src/address/address.service.ts:51]: Why don't fetch it in data fetcher?

File: packages/data-fetcher/src/rpcProvider/webSocketProviderExtended.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { Logger } from "@nestjs/common";
```
Comment (Romsters) [inline packages/data-fetcher/src/rpcProvider/webSocketProviderExtended.ts:1]: Worth to replace it with the latest provider version

File: packages/worker/src/entities/addressTransaction.entity.ts [hunk 4]
```diff
@@ -30,8 +31,8 @@ export class AddressTransaction extends BaseEntity {
   @Column({ type: "bigint", transformer: bigIntNumberTransformer })
   public readonly blockNumber: number;
 
-  @Column({ type: "timestamp" })
-  public readonly receivedAt: Date;
+  @Column({ type: "timestamp", transformer: stringDateTransformer })
+  public readonly receivedAt: string;
```
Comment (Romsters) [inline packages/worker/src/entities/addressTransaction.entity.ts:35]: What's the reason for this?

File: packages/worker/src/entities/addressTransfer.entity.ts [hunk 5]
```diff
@@ -39,16 +39,16 @@ export class AddressTransfer extends BaseEntity {
...
+  public readonly timestamp: string;
 
   @Column({ type: "enum", enum: TokenType, default: TokenType.ETH })
   public readonly tokenType: TokenType;
 
   @Column({ type: "boolean" })
   public readonly isFeeOrRefund: boolean;
 
-  @Column({ type: "jsonb", nullable: true, transformer: transferFieldsTransformer })
+  @Column({ type: "jsonb", nullable: true })
```
Comment (Romsters) [inline packages/worker/src/entities/addressTransfer.entity.ts:51]: Why is transferFieldsTransformer removed?

File: packages/worker/src/entities/balance.entity.ts [hunk 6]
```diff
@@ -22,6 +20,6 @@ export class Balance extends BaseEntity {
   @PrimaryColumn({ type: "bigint", transformer: bigIntNumberTransformer })
   public readonly blockNumber: number;
 
-  @Column({ type: "varchar", length: 128, transformer: bigNumberTransformer })
-  public readonly balance: BigNumber;
+  @Column({ type: "varchar", length: 128 })
+  public readonly balance: string;
```
Comment (Romsters) [inline packages/worker/src/entities/balance.entity.ts:24]: Why is this changed to string?

File: packages/worker/src/rpcProvider/webSocketProviderExtended.ts [hunk 7]
```diff
@@ -1,73 +1,18 @@
+import { Logger } from "@nestjs/common";
 import { providers } from "ethers";
```
Comment (Romsters) [inline packages/worker/src/rpcProvider/webSocketProviderExtended.ts:2]: I'd rather keep the latest version

File: packages/data-fetcher/README.md (hunk 1 above)
Comment (vasyl-ivanchuk) [reply packages/data-fetcher/README.md:1]: Agreed, I want to test everything first and then add docs in the following PR.

File: packages/worker/src/rpcProvider/webSocketProviderExtended.ts (hunk 7 above)
Comment (vasyl-ivanchuk) [reply packages/worker/src/rpcProvider/webSocketProviderExtended.ts:2]: Do you think it's worth keeping it considering that we need to cover it with tests? I was thinking about removing it in both worker and data-fetcher.

File: packages/worker/src/entities/addressTransaction.entity.ts (hunk 4 above)
Comment (vasyl-ivanchuk) [reply packages/worker/src/entities/addressTransaction.entity.ts:34]: Removed `stringDateTransformer`.

**Review by Romsters — APPROVED** [review]
